MvvmCross/ViewModels/MvxDefaultViewModelLocator.cs

```diff
@@ -100,20 +100,20 @@ protected void RunViewModelLifecycle(IMvxViewModel viewModel, IMvxBundle paramet
             try
             {
                 CallCustomInitMethods(viewModel, parameterValues);
-                if (navigationArgs.Cancel)
+                if (navigationArgs?.Cancel == true)
```

Comment:
I don't think the `navigationArgs` will all of the sudden be null after this point. So it should be enough just to check for null one place.

Reply:
That doesn't appear to be the case, at least on Android. Stepping through Playground.Droid in the debugger I see the `RunViewModelLifecycle(IMvxViewModel viewModel, IMvxBundle parameterValues, IMvxBundle savedState, IMvxNavigateEventArgs navigationArgs)` is called first and `navigationArgs` is null. Then `RunViewModelLifecycle<TParameter>(IMvxViewModel<TParameter> viewModel, TParameter param, IMvxBundle parameterValues, IMvxBundle savedState, IMvxNavigateEventArgs navigationArgs)` is called later and `navigationArgs` is null there as well. `navigationArgs` is currently always null during both of these methods. Removing any of these null checks I've added in this PR just moves where the NullReferenceException originates.

There seems to be a bug in PR #3046 where `navigationArgs` are not being supplied in all scenarios so the default value of null is sometimes used. There were no null checks against `navigationArgs` even though it defaults to null. This PR addresses the lack of null checks and allows Playground to run again. There is still an underlying problem introduced by #3046 if `navigationArgs` isn't expected to be null.

Reply:
What I mean is not that the argument cannot be null. But rather inside of `RunViewModelLifecycle` there is nothing that modifies `navigationArgs` and it should be fine checking it only once.